napari/components/viewer_model.py

```diff
@@ -1000,6 +1000,73 @@ def add_vectors(
         self.add_layer(layer)
         return layer
 
+    def _add_layer_from_data(
```

Comment:
I don't like `_add_layer_from_data` as we use `data` in other ways in other places (i.e. just as the array). I don't think `_add_layer_from_tuple` or `_add_layer_from_dict` are any better, so I'm not quite sure right now. I will sleep on it and maybe have a better idea in the morning :-)